File: dbflow/src/main/java/com/raizlabs/android/dbflow/structure/database/BaseDatabaseStatement.java
package com.raizlabs.android.dbflow.structure.database;

import androidx.annotation.Nullable;

/**
 * Description: Default implementation for some {@link DatabaseStatement} methods.
 */
public abstract class BaseDatabaseStatement implements DatabaseStatement {

    @Override
    public void bindStringOrNull(int index, @Nullable String s) {
        if (s != null) {
            bindString(index, s);
        } else {
            bindNull(index);
        }
    }

    @Override
    public void bindNumber(int index, @Nullable Number number) {
        bindNumberOrNull(index, number);
    }

    @Override
    public void bindNumberOrNull(int index, @Nullable Number number) {
        if (number != null) {
            bindLong(index, number.longValue());
        } else {
            bindNull(index);
        }
    }

    @Override
    public void bindDoubleOrNull(int index, @Nullable Double aDouble) {
        if (aDouble != null) {
            bindDouble(index, aDouble);
        } else {
            bindNull(index);
        }
    }

    @Override
    public void bindFloatOrNull(int index, @Nullable Float aFloat) {
        if (aFloat != null) {
            bindDouble(index, aFloat);
        } else {
            bindNull(index);
        }
    }

    @Override
    public void bindBlobOrNull(int index, @Nullable byte[] bytes) {
        if (bytes != null) {
            bindBlob(index, bytes);
        } else {
            bindNull(index);
        }
    }

}

File: lib/src/main/kotlin/com/dbflow5/database/BaseDatabaseStatement.kt
package com.dbflow5.database

/**
 * Description: Default implementation for some [DatabaseStatement] methods.
 */
abstract class BaseDatabaseStatement : DatabaseStatement {

    override fun bindStringOrNull(index: Int, s: String?) {
        if (s != null) {
            bindString(index, s)
        } else {
            bindNull(index)
        }
    }

    override fun bindNumber(index: Int, number: Number?) {
        bindNumberOrNull(index, number)
    }

    override fun bindNumberOrNull(index: Int, number: Number?) {
        if (number != null) {
            bindLong(index, number.toLong())
        } else {
            bindNull(index)
        }
    }

    override fun bindLongOrNull(index: Int, aLong: Long?) {
        if (aLong != null) {
            bindLong(index, aLong)
        } else {
            bindNull(index)
        }
    }

    override fun bindDoubleOrNull(index: Int, aDouble: Double?) {
        if (aDouble != null) {
            bindDouble(index, aDouble)
        } else {
            bindNull(index)
        }
    }

    override fun bindFloatOrNull(index: Int, aFloat: Float?) {
        if (aFloat != null) {
            bindDouble(index, aFloat.toDouble())
        } else {
            bindNull(index)
        }
    }

    override fun bindBlobOrNull(index: Int, bytes: ByteArray?) {
        if (bytes != null) {
            bindBlob(index, bytes)
        } else {
            bindNull(index)
        }
    }

}
